### PR TITLE
Pin 1801: add documents endpoint

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -505,6 +505,231 @@ paths:
...
+              description: Max allowed requests within time interval
+            'X-Rate-Limit-Remaining':
+              schema:
+                type: integer
+              description: Remaining requests within time interval
+            'X-Rate-Limit-Interval':
+              schema:
+                type: integer
+              description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+  '/agreements/{agreementId}/consumer-documents/{documentId}':
```
Comment (galales) [inline src/main/resources/interface-specification.yml:623]: missing rate limiting headers and 429 status for below endpoints

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala [hunk 2]
```diff
@@ -310,4 +321,109 @@ final case class AgreementsApiServiceImpl(
...
+  ): Route = {
+    logger.info(s"Deleting consumer document $documentId for agreement $agreementId")
+
+    val result: Future[HttpEntity.Strict] =
+      for {
+        uuid       <- agreementId.toFutureUUID
+        agreement  <- agreementProcessService.getAgreementById(uuid)
+        contract   <- agreement.contract.toFuture(ContractNotFound(agreementId))
+        byteStream <- fileManager.get(ApplicationConfiguration.storageContainer)(contract.path)
+      } yield HttpEntity(ContentType(MediaTypes.`application/pdf`), byteStream.toByteArray())
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala]: this endpoint should delete the document but it is actually a retrieve

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala (hunk 2 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala]: +1

**Review by TonioGela — APPROVED** [review]

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -505,6 +505,250 @@ paths:
...
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/Problem'
+        '404':
+          description: Resource Not Found
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/Problem'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: rate limiting headers missing

File: src/main/resources/interface-specification.yml [hunk 4]
```diff
@@ -505,6 +505,250 @@ paths:
...
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: rate limiting headers missing

File: src/main/resources/interface-specification.yml [hunk 5]
```diff
@@ -505,6 +505,250 @@ paths:
...
+        '429':
+          description: Too Many Requests
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/Problem'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: rate limiting headers missing

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala [hunk 6]
```diff
@@ -310,4 +321,109 @@ final case class AgreementsApiServiceImpl(
...
+  ): Route = {
+    logger.info(s"Deleting consumer document $documentId for agreement $agreementId")
+
+    val result: Future[Unit] =
+      for {
+        uuid      <- agreementId.toFutureUUID
+        agreement <- agreementProcessService.getAgreementById(uuid)
+        contract  <- agreement.contract.toFuture(ContractNotFound(agreementId))
+        _         <- fileManager.delete(ApplicationConfiguration.storageContainer)(contract.path)
+      } yield ()
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala]: no calls to `agreementProcessService.deleteConsumerDocument` 👀

**Review by galales — CHANGES_REQUESTED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**